whisper.cpp

```diff
@@ -118,6 +118,9 @@ static void byteswap_tensor(ggml_tensor * tensor) {
 // logging
 //
 
+#ifdef WHISPER_LOGCAT
+#include <android/log.h>
+#endif
```

Comment:
To be honest, I'm not convinced that including `<android/log.h>` in whisper.cpp is the best approach. It seems more practical to use a callback instead. 

I'm also wondering why do we continue to use `WHISPER_PRINT_DEBUG` when we already have `WHISPER_LOG`? Couldn't we just create a function named `WHISPER_LOG_DEBUG` that offers both console output and callback functionality? @ggerganov

Reply:
Well, I see the callback logic just now,  I agree with you, it's the right way!

Reply:
Yup, logcat is better to not be part of `whisper.cpp` - utilize the log callback

Regarding `WHISPER_PRINT_DEBUG` - it's used for debugging that I don't think will be really useful to the "outside world". But we could add `WHISPER_LOG_DEBUG` instead for consistency. Probably have to extend `enum ggml_log_level` with new `GGML_LOG_LEVEL_DEBUG` value